src/hipscat_import/runtime_arguments.py

```diff
@@ -0,0 +1,115 @@
+        if self.dask_threads_per_worker <= 0:
+            raise ValueError("dask_threads_per_worker should be greater than 0")
+
+        if not file_io.does_file_or_directory_exist(self.output_path):
+            raise FileNotFoundError(
+                f"output_path ({self.output_path}) not found on local storage"
+            )
+
+        self.catalog_path = file_io.append_paths_to_pointer(
+            self.output_path, self.output_catalog_name
+        )
+        if not self.overwrite:
+            if file_io.directory_has_contents(self.catalog_path):
+                raise ValueError(
+                    f"output_path ({self.catalog_path}) contains files."
+                    " choose a different directory or use --overwrite flag"
+                )
+        file_io.make_directory(self.catalog_path, exist_ok=True)
+
+        if self.tmp_dir:
+            if not file_io.does_file_or_directory_exist(self.tmp_dir):
+                raise FileNotFoundError(
+                    f"tmp_dir ({self.tmp_dir}) not found on local storage"
+                )
+            self.tmp_path = file_io.append_paths_to_pointer(
+                self.tmp_dir, self.output_catalog_name, "intermediate"
+            )
+        elif self.dask_tmp:
+            if not file_io.does_file_or_directory_exist(self.dask_tmp):
+                raise FileNotFoundError(
+                    f"dask_tmp ({self.dask_tmp}) not found on local storage"
+                )
+            self.tmp_path = file_io.append_paths_to_pointer(
+                self.dask_tmp, self.output_catalog_name, "intermediate"
+            )
+        else:
+            self.tmp_path = file_io.append_paths_to_pointer(
+                self.catalog_path, "intermediate"
+            )
+        file_io.make_directory(self.tmp_path, exist_ok=True)
+
+    def provenance_info(self) -> dict:
+        """Fill all known information in a dictionary for provenance tracking.
+
+        Returns:
+            dictionary with all argument_name -> argument_value as key -> value pairs.
+        """
+        runtime_args = {
+            "catalog_name": self.output_catalog_name,
+            "output_path": str(self.output_path),
+            "output_catalog_name": self.output_catalog_name,
+            "tmp_dir": str(self.tmp_dir),
+            "overwrite": self.overwrite,
+            "dask_tmp": str(self.dask_tmp),
+            "dask_n_workers": self.dask_n_workers,
+            "dask_threads_per_worker": self.dask_threads_per_worker,
+            "catalog_path": self.catalog_path,
+            "tmp_path": str(self.tmp_path),
+        }
+
+        runtime_args.update(self.additional_provenance_info())
+        provenance_info = {
+            "tool_name": "hipscat_import",
+            "version": version("hipscat-import"),
+            "runtime_args": runtime_args,
+        }
+
+        return provenance_info
+
+    def additional_provenance_info(self):
```

Comment:
I think this method name is slightly confusing given that the additional info is added to the runtime args in the provenance info and not the provenance info itself

Reply:
Changed to `additional_runtime_provenance_info`. LMK if you think of a better name.